### PR TITLE
Cycle detection using BFS in C++

### PR DESCRIPTION
**Pull request by garvitgupta13**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
YES

### Description
Add a cpp code to detect cycle in an undirected graph using BFS. The current detect cycle code uses dfs and I want to add extra approach using BFS

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#7415

### CODE REVIEW

**Review by ankitaggarwal23 — COMMENTED** [review]
Code already present

**Comment by garvitgupta13** [comment]
> Code already present

Hey @ankitaggarwal23 the PR #1157 contains the code for checking a cycle using DFS, I have added another approach which checks the cycle using BFS
In the description also I have described about this

**Comment by garvitgupta13** [comment]
> Code already present

The code in detect_cycle.cpp and checkCycle_bfs.cpp is entirely different and the algorithm used is also different.

File: C-Plus-Plus/graphs/checkCycle_bfs.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
+// Cycle Detection in Undirected Graph using BFS
+#include <bits/stdc++.h>
+using namespace std;
+
+vector<int> arr[100000];    //Adjacency List storing the graph
+vector<int> vis(100000, 0); //Visited array, it stores wheather the node is visited or not
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/graphs/checkCycle_bfs.cpp]: Shift comments above the lines of code.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Rest looks good to merge

File: C-Plus-Plus/graphs/checkCycle_bfs.cpp (hunk 1 above)
Comment (garvitgupta13) [reply C-Plus-Plus/graphs/checkCycle_bfs.cpp]: done

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**